SalesforceCommerceCloud/pwa-kit
# useBasket hooks integration by page

# Description

This PR demonstrates we can integrate Commerce-sdk-react page by page. This POC provided two examples on PDP and cart page. 
- On PDP, it can render PDP data by using useProduct, and useCategory. It can add an item to basket using mutation hooks like `useCustomerBaskets` and `const addItemToBasket = useShopperBasketsMutation('addItemToBasket')`
- On Cart page, it can fetch basket data using `useCustomerBaskets` and fetching productDetails using `useProducts`
- On Home page, a simple replacement of getProps, and usage of useProduct hook




# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Pull the code
- Rebuild your commerce-sdk-react
- Start Retail App
- Test that you can add an item and the basket in the header is updated accordling. You can go to the cart page and see the items in the basket

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
